### DISCUSSION TITLE
Long load time to view recordings?

### DISCUSSION THREAD

**Discussion by eblohm** [blakeblackshear/frigate#7809] [Ask A Question] [answered]
Sorry if this isn't the right way to post this question but I'm going to be upgrading my server this weekend and am considering adding more RAM or upgrading my CPU and wanted to get some input on an "issue" that I've had.

For most of my cameras I store 30 days of recordings. I've noticed that a lot of the time when I want to go back and check something out, it takes a while for the recordings to load (like over a minute, or sometimes it doesn't load at all), but occasionally it'll be instant.

Since I'm going to be upgrading other parts of my server this weekend I wanted to see if there was anything I could do to improve the speed they load? I've got 16GB of DDR4 RAM (which is typically at 65% utilized by my server) and an i5-11600K. When I try to view recordings and watch the server logs it looks like my CPU is what spikes in usage pretty significantly so I'm assuming if there's a part I can upgrade to improve the load times, it's the CPU. I'm using a WD Purple 4TB hard drive to record to.

If it _is_ the CPU that needs to upgrade to help this out I may just deal with it because I'm not sure it's worth spending hundreds for a minor inconvenience, but if it's the RAM I don't have too much of an issue spending to add another 16GB (to bring me to 32GB). Does anyone have any suggestions to improve the load time of recordings?

**Comment by NickM-27** [comment]
What is your config and docker compose?

The most common reason this happens is because the database is stored on slow storage (HDD or network share), is fragmented and full (in this case should run `VACUUM;` on the db), or it can be due to slow storage on the volume holding the recordings directory

to be clear slow does not necessarily mean the hardware is slow, sometimes it just means bad mounting params are being used.

I'd start with including the config to confirm the first set of issues are not the case

**Reply by eblohm** [reply]
I'm using unRAID so I don't really have a docker compose outside of whatever the basic unRAID template is.

Here's a pastebin of my config - https://pastebin.com/4Xa4jf98

My Frigate drive is a 4TB WD Purple @ 5,400 RPM. It typically has between 200-230GB free so it's pretty full. I've been debating buying the 8TB Purple drive so I can add a few more cameras and that's a 7,200 RPM drive but wasn't planning on that yet, but I guess if it'd help out with the speed of recordings loading maybe I'll take care of that this weekend instead of upgrading the actual hardware

**Reply by NickM-27** [reply]
Unraid just uses docker CLI. It shows the CLI command when you apply the config. 

The easiest way is to just configure frigate to store the database in /config instead.

**Reply by eblohm** [reply]
Here's the config that gets applied when I make changes to the container - https://pastebin.com/W4HhVrHC

**Reply by NickM-27** [reply, marked as answer]
Cool, so if you move the database files to the frigate appdata folder and add

```
database:
  path: /config/frigate.db
```

To the frigate config it will hopefully perform better.

**Reply by eblohm** [reply]
Awesome, just tried it. Looks like it cleared my recordings (not a big deal) but the few minutes that have been recorded load almost instantly. Thank you!

**Reply by NickM-27** [reply]
That shouldn't be the case, nothing should be lost, so it must've not been moved to the correct location.

**Reply by eblohm** [reply]
OH! I totally misunderstood what you were saying, I just updated it to where _my_ `frigate.db` file is, not just `/config/frigate.db`. Now it works, I see all the recordings and it's loading super fast. Thank you!!
